Adapted from ../linux-samsung-dream/05-BGR-to-RGB-colors.patch.
Fixes color order - from BGRA to RGBA.

diff --git a/drivers/video/fbdev/exynos/dpu_9810/decon_dsi.c b/drivers/video/fbdev/exynos/dpu_9810/decon_dsi.c
index 07a084d805af..4504591db73e 100644
--- a/drivers/video/fbdev/exynos/dpu_9810/decon_dsi.c
+++ b/drivers/video/fbdev/exynos/dpu_9810/decon_dsi.c
@@ -807,11 +807,11 @@ int decon_check_var(struct fb_var_screeninfo *var, struct fb_info *info)
 	case 24:
 		/* our 24bpp is unpacked, so 32bpp */
 		var->bits_per_pixel	= 32;
-		var->red.offset		= 16;
+		var->red.offset		= 0;
 		var->red.length		= 8;
 		var->green.offset	= 8;
 		var->green.length	= 8;
-		var->blue.offset	= 0;
+		var->blue.offset	= 16;
 		var->blue.length	= 8;
 		break;
 
